src/aiq/builder/workflow_builder.py

```diff
@@ -601,6 +562,26 @@ async def get_retriever_config(self, retriever_name: str | RetrieverRef) -> Retr
     def get_user_manager(self):
         return UserManagerHolder(context=AIQContext(self._context_state))
 
+    async def add_telemetry_exporter(self, name: str, config: TelemetryExporterBaseConfig) -> None:
+        """Add an configured telemetry exporter to the builder.
+
+        Args:
+            name (str): The name of the telemetry exporter
+            config (TelemetryExporterBaseConfig): The configuration for the exporter
+        """
+        if (name in self._telemetry_exporters):
+            raise ValueError(f"Telemetry exporter '{name}' already exists in the list of telemetry exporters")
+
+        exporter_info = self._registry.get_telemetry_exporter(type(config))
+
+        # Build the exporter outside the lock (parallel)
+        exporter_context_manager = exporter_info.build_fn(config, self)
+
+        # Only protect the shared state modifications (serialized)
+        async with self._telemetry_exporters_lock:
+            exporter = await self._get_exit_stack().enter_async_context(exporter_context_manager)
+            self._telemetry_exporters[name] = ConfiguredTelemetryExporter(config=config, instance=exporter)
+
```

Comment:
I dont think this is doing what you think it should. Unless you launch these as a task, they wont start right away. Unclear how to make it operate in parallel